chalice/deploy/planner.py

```diff
@@ -159,6 +178,180 @@ def _add_result_to_plan(self,
+                models.StoreMultipleValue(
+                    name=variable_name,
+                    value=[Variable('base_path_mapping')]
+                ),
+                models.RecordResourceVariable(
+                    resource_type='domain_name',
+                    resource_name=resource.domain_name.resource_name,
+                    name='base_path_mappings',
+                    variable_name=variable_name
+                ),
+            ])
+        else:
+            deployed = self._remote_state.resource_deployed_values(
+                resource.domain_name
+            )
+            for base_path_mapping in deployed['base_path_mappings']:
+
+                base_path = base_path_mapping['key'].lstrip('/')
+                if not base_path:
+                    base_path = '(none)'
+                if base_path != resource.base_path:
+                    continue
+
+                api_calls.extend([
+                    models.StoreMultipleValue(
+                        name=variable_name,
+                        value=[base_path_mapping]
+                    ),
+                    models.RecordResourceVariable(
+                        resource_type='domain_name',
+                        resource_name=resource.domain_name.resource_name,
+                        name='base_path_mappings',
+                        variable_name=variable_name
+                    ),
+                ])
+        return api_calls
+
+    def _plan_domainname(self, resource):
+        # type: (models.DomainName) -> Sequence[InstructionMsg]
+        api_calls = []  # type: List[InstructionMsg]
+
+        params = {
+            'domain_name': resource.domain_name,
+            'endpoint_configuration': resource.endpoint_configuration,
+            'security_policy': resource.security_policy,
+            'certificate_arn': resource.certificate_arn,
+            'regional_certificate_arn': resource.regional_certificate_arn
+        }
+        if resource.hosted_zone_id:
+            params['hosted_zone_id'] = resource.hosted_zone_id
+
+        if not self._remote_state.resource_exists(resource):
+            params['protocol'] = resource.protocol
+            params['tags'] = resource.tags
+            api_calls.extend([
+                (models.APICall(
+                    method_name='create_domain_name',
+                    params=params,
+                    output_var=resource.resource_name
+                ), "Creating custom domain name: %s\n" % resource.domain_name),
+                models.StoreValue(
+                    name='hosted_zone_id',
+                    value=KeyDataVariable(resource.resource_name,
+                                          'hosted_zone_id')
+                ),
+                models.RecordResourceVariable(
+                    resource_type='domain_name',
+                    resource_name=resource.resource_name,
+                    name='hosted_zone_id',
```

Comment:
I think it would be useful to update the `DeploymentReporter` to display the hosted_zone_id/domain_name or the distributionDomainName when the app is deployed.  The users is going to need these values to finish setting up their custom domain, and except for hosted zone id, we don't store the other values in `.chalice/deployed/dev.json`, so they'd have to use the CLI/console to find those values.